crates/ratings_new/src/grpc/user.rs

```diff
@@ -0,0 +1,235 @@
+// FIXME: Remove these dependencies
+use ratings::features::user::entities::{User as OldUser, Vote as OldVote};
+use sqlx::PgConnection;
+
+use crate::{context::Claims, ratings::{categories::update_categories, users::{create_or_seen_user, delete_user_by_client_hash}, votes::{find_user_votes, get_snap_votes_by_client_hash, save_vote_to_db}}, Context};
+use crate::proto::user::{
+    user_server::{User, UserServer},
+    AuthenticateRequest, AuthenticateResponse, GetSnapVotesRequest, GetSnapVotesResponse,
+    ListMyVotesRequest, ListMyVotesResponse, Vote, VoteRequest,
+};
+use time::OffsetDateTime;
+use tonic::{Request, Response, Status};
+use tracing::{error, warn};
+
+/// The length we expect a client hash to be, in bytes
+pub const EXPECTED_CLIENT_HASH_LENGTH: usize = 64;
+
+/// An empty struct used to construct a [`UserServer`]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct UserService;
+// Store jwt encoder here. If needed in multiple places, arc it.
+
+impl UserService {
+    /// The paths which are accessible without authentication, if any
+    pub const PUBLIC_PATHS: [&'static str; 2] = [
+        "ratings.features.user.User/Register",
+        "ratings.features.user.User/Authenticate",
+    ];
+
+    /// Converts this service into its corresponding server
+    pub fn to_server(self) -> UserServer<UserService> {
+        self.into()
+    }
+}
+
+impl From<UserService> for UserServer<UserService> {
+    fn from(value: UserService) -> Self {
+        UserServer::new(value)
+    }
+}
+
+#[tonic::async_trait]
+impl User for UserService {
+    #[tracing::instrument(level = "debug")]
+    async fn authenticate(
+        &self,
+        mut request: Request<AuthenticateRequest>,
+    ) -> Result<Response<AuthenticateResponse>, Status> {
+        let ctx = request
+            .extensions_mut()
+            .remove::<Context>()
+            .expect("Expected AppContext to be present");
+        // TODO: is there where we expect the pg_connection?
+        let mut conn = request.extensions_mut().remove::<PgConnection>().expect("Expected PgConnection to be present");
+        let AuthenticateRequest { id } = request.into_inner();
+
+        if id.len() != EXPECTED_CLIENT_HASH_LENGTH {
+            let error = format!(
+                "Client hash must be of length {:?}",
+                EXPECTED_CLIENT_HASH_LENGTH,
+            );
+            return Err(Status::invalid_argument(error));
+        }
+        
+        // FIXME: replace with new struct
+        let user = OldUser::new(&id);
```

Comment:
@sminez I was not 100% sure what we wanted to do with entities from `crates/ratings/src/features/*/entities`, so I still make use of them. Will they end up in `crates/ratings_new/src/ratings`? Would you like me to move them over?